### PR TITLE
SCANNPM-9 Allow to override the SonarQube version for internal testing

### PR DESCRIPTION
**Pull request by 7PH**
As per the specification

### CODE REVIEW

File: src/java.ts [hunk 1]
```diff
@@ -18,18 +18,17 @@
...
+import fs from 'fs';
+import * as fsExtra from 'fs-extra';
+import path from 'path';
+import semver, { SemVer } from 'semver';
 import * as stream from 'stream';
 import tarStream from 'tar-stream';
 import { promisify } from 'util';
-import semver, { SemVer } from 'semver';
-import { log, LogLevel } from './logging';
+import zlib from 'zlib';
```
Comment (lucas-paulger-sonarsource) [inline src/java.ts:31]: not for this PR, but lets align on a formatting ruleset we can enforce in the CI / locally to reduce unrelated changes (I like your setting to order alphabetically)
I created a subtask for this
https://sonarsource.atlassian.net/browse/SCANNPM-16

File: src/types.ts [hunk 2]
```diff
@@ -54,6 +54,7 @@ export enum ScannerProperty {
   SonarScannerSonarCloudURL = 'sonar.scanner.sonarcloudUrl',
   SonarScannerJavaExePath = 'sonar.scanner.javaExePath',
   SonarScannerWasEngineCacheHit = 'sonar.scanner.wasEngineCacheHit',
+  SonarScannerInternalSqVersion = 'sonar.scanner.internal.sqVersion',
```
Comment (lucas-paulger-sonarsource) [inline src/types.ts:57]: for backwards compatability we would also need to check for `sonar.scanner.versionSimulation`

**Review by lucas-paulger-sonarsource — APPROVED** [review]
one small callout

File: src/types.ts (hunk 2 above)
Comment (7PH) [reply src/types.ts:57]: I don't see that the current version of NPM scanner supports `versionSimulation`, am I missing something here?

**Pull request merged**